scripts/generate-wix-module-list.js

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
+const fs = require("fs-extra");
+const _ = require("lodash");
+const { corvidDir, createTsProgram } = require("./util");
+
+const MODULE_LIST_FILE_NAME = "wixModules.json";
+const TS_CONFIG = {
+  files: ["types/common/declaration.d.ts"]
+};
+
+function generateModuleList() {
+  let tmpConfigPath = `${corvidDir}/tsconfig.wix-api.json`;
+  fs.writeJson(tmpConfigPath, TS_CONFIG)
```

Comment:
why the need of writing a file only to delete it later? Is that because that's how a typescript `Program` expects it?

Reply:
No.
I kinda didn't wan't to cause confusion between this process then the typings file generation process.
the typings-file configs are a declaration of what will eventually go into the final public typings and so they are a public commitment and should be formally declared.
The module list on the other hand is derived from the common/declaration.d.ts file, os the file need not to be declared, I could have set a config file like the other, but as I said I didn't want to create  confusion &&/|| mess

Reply:
WDYM by the &&/||? not sure I follow.
Why would it cause confusion? You are not generating this file here, you are using this as the base for creating the module list. I think creating an additional config file (only to remove it later) is adding confusion, not reducing it. If you wanted to *actually* create this file (not as part of a temp file, but rather actually create this file and commit it to git) then I think your argument would make more sense.

Reply:
&&/|| = And/Or
What I was trying to say is that the other config file are a form of declaration, while this one would/is a simple tool and has no declaratory meaning.
If you are reading through code and the logical process I agree it adds complexity, not decrease it.
But when I take a look at the project itself I know what job the config files have, and it is not the the same with this config (tmp)file and so not putting it on the same level of these file decrease confusion